WriteHDLUtils.py

```diff
@@ -893,17 +908,48 @@ def writeModuleInst_generic(module, hls_src_dir, f_writeTemplatePars,
                     # Add the memory instance to the port string
                     # Assumes a sorted memModuleList due to arrays?
                     if portname.find("in") != -1:
-                        string_mem_ports += writeProcMemoryRHSPorts(tmp_argname,memory)
+                        if isinstance(memory, list):
```

Comment:
Please correct me if I'm wrong, is the reason for having all of these if-statements because labelConnectedMemoryArrays/getListsOfGroupedMemories puts some of the memories in a list? (stubPairs, innerStubs, outerStubs) If yes, would it work to uncomment labelConnectedMemoryArrays and remove all the ifs similar to this?